libutils/string_lib.c

```diff
@@ -372,13 +372,140 @@ bool EmptyString(const char *s)
 
 /*********************************************************************/
 
-long StringToLong(const char *str)
+/**
+ * @brief Converts a string of numerals in base 10 to a long integer.
+ *
+ * Result is stored in *number, return value should be chacked.
+ * On error *number is unmodified and -1 is returned.
```

Comment:
Not -1, but "an error code is returned".

I'm not sure if this is the best way though. You might as well return bool type and `false` on any error. Still I'm split on this, so error code is OK.

Reply:
I like int better because it's easier to debug, also makes it easier to check for integer overflow for example.